packages/office-ui-fabric-react/src/components/Facepile/FacepileResponsive.tsx

```diff
@@ -0,0 +1,139 @@
+import * as React from 'react';
+import {
+  autobind,
+  BaseComponent,
+  buttonProperties,
+  css,
+  divProperties,
+  getId,
+  getNativeProps
+} from '../../Utilities';
+import {
+  Facepile
+} from './Facepile';
+import {
+  IFacepileProps,
+  IFacepilePersona,
+  OverflowButtonType
+} from './Facepile.Props';
+import {
+  FocusZone,
+  FocusZoneDirection
+} from '../../FocusZone';
+import {
+  BaseButton,
+  DefaultButton
+} from '../../Button';
+import {
+  Icon
+} from '../../Icon';
+import {
+  PersonaCoin,
+  PersonaSize
+} from '../../PersonaCoin';
+import {
+  ResizeGroup
+} from '../../ResizeGroup';
+import * as stylesImport from './Facepile.scss';
+const styles: any = stylesImport;
+
+interface IOverflowData {
+  props: IFacepileProps;
+  primary: IFacepilePersona[];
+  overflow: IFacepilePersona[];
+  cacheKey?: string;
+}
+
+export interface IFacepileResponsiveProps extends IFacepileProps {
+  /**
+   * Set number of pixel to use for width instead of calculating it internally.
+   */
+  width?: number;
+}
+
+export class FacepileResponsive extends BaseComponent<IFacepileResponsiveProps, {}> {
+  public static defaultProps: IFacepileResponsiveProps = {
+    maxDisplayablePersonas: 5,
+    personas: [],
+    personaSize: PersonaSize.extraSmall
+  };
+
+  public render(): JSX.Element {
+    let {
+      ariaDescription,
+      chevronButtonProps,
+      maxDisplayablePersonas,
+      overflowButtonProps,
+      overflowButtonType,
+      className,
+      personas,
+      showAddButton,
+      width
+    } = this.props;
+
+    let facepileWrapperProps = {
+      width: width ? `${width}px` : 'auto'
+    };
+
+    let numPersonasToShow: number = Math.min(personas.length, maxDisplayablePersonas || personas.length);
+
+    let facepileData: IOverflowData = {
+      props: this.props,
+      primary: personas.slice(0, numPersonasToShow),
+      overflow: personas.slice(numPersonasToShow)
+    };
+
+    return (
+      <div style={ { ...facepileWrapperProps } }>
+        <ResizeGroup
+          className={ className }
+          data={ facepileData }
+          onReduceData={ this._showLess }
+          onGrowData={ this._showMore }
+          onRenderData={ (data: IOverflowData) => {
+            return (
+              <Facepile {...this.props}
+                maxDisplayablePersonas={ data.primary.length }
```

Comment:
I don't love this.  Input would be nice.  Should Facepile take an overflow?

Reply:
Shouldn't facepile just take in the exact faces to render in the top and the overflow? Or at least the top level count?

Reply:
Probably worth aligning with the pattern established by command bar + overflow set. I can't recall what they did.

Reply:
@christiango Matt and I just had a long discussion today about this....seems I totally missed this comment :D

I agree, I like the two array approach of main and overflow. We could consider deprecating maxDisplayedPersonas, and moving the the two array approach, making maxDisplayedPersonas and overflowPersonas mutually exclusive.